bundles/admin/admin-hierarchical-layerlist/components/Group.js

```diff
@@ -0,0 +1,258 @@
+                    dialog.fadeout(5000);
+                    return;
+                }
+
+                var data = {
+                    id: id
+                };
+
+                // Confirm dialog
+                var confirmDialog = Oskari.clazz.create('Oskari.userinterface.component.Popup');
+                var confirmBtnOk = Oskari.clazz.create('Oskari.userinterface.component.buttons.OkButton');
+                var confirmBtnCancel = Oskari.clazz.create('Oskari.userinterface.component.buttons.CancelButton');
+
+                confirmBtnOk.addClass('primary');
+                confirmBtnOk.setHandler(function() {
+                    confirmDialog.close();
+                    me._deleteGroup(data, popup, opts.type, tool);
+                });
+
+                confirmBtnCancel.setHandler(function() {
+                    confirmDialog.close();
+                });
+
+                var groupName = Oskari.getLocalized(group.getName());
+
+                confirmDialog.show(me.locale('confirms.groupDelete.title'), me.locale('confirms.groupDelete.message', {
+                    groupname: groupName
+                }), [confirmBtnCancel, confirmBtnOk]);
+                confirmDialog.makeModal();
+
+            });
+            returnObject.buttons.push(btnDelete);
+        }
+
+        var btnCancel = Oskari.clazz.create('Oskari.userinterface.component.buttons.CancelButton');
+        btnCancel.addClass('cancel');
+        btnCancel.setHandler(function() {
+            popup.close();
+            tool.removeClass('active');
+        });
+        returnObject.buttons.push(btnCancel);
+        var btnOk = Oskari.clazz.create('Oskari.userinterface.component.Button');
+        returnObject.buttons.push(btnOk);
+        popup.addClass('admin-hierarchical-layerlist-group');
+
+        var loc = (Oskari.getMsg('DivManazer', 'LanguageSelect').languages) ? Oskari.getMsg('DivManazer', 'LanguageSelect').languages : {};
+
+        var selectableGroup = Oskari.clazz.create('Oskari.userinterface.component.CheckboxInput');
+        selectableGroup.addClass('selectable-group');
+        selectableGroup.setTitle(me.locale('selectableGroup'));
+        selectableGroup.setChecked(!!opts.selectable);
+
+        btnOk.addClass('add');
+        var btnOkLocale = (id) ? me.locale('buttons.update') : me.locale('buttons.add');
+        btnOk.setTitle(btnOkLocale);
+        btnOk.setHandler(function() {
+            var data = {
+                locales: {},
+                selectable: selectableGroup.isChecked(),
+                parentId: parentId,
+                id: id
+            };
+            var hasValidLocales = true;
+            popup.getJqueryContent().find('.oskari-textinput.group-name').each(function() {
+                var el = jQuery(this);
+
+                var value = el.find('input').val().trim();
+                data.locales[el.attr('data-locale')] = value;
+                if (value.length < 4) {
```

Comment:
The minimum allowed length for trimmed group name is 4 characters? Is this some requirement for something? Isn't a single character enough?